plugins/net_plugin/net_plugin.cpp

```diff
@@ -3420,14 +3442,11 @@ namespace eosio {
 
    // called from application thread
    void net_plugin_impl::transaction_ack(const std::pair<fc::exception_ptr, packed_transaction_ptr>& results) {
-      dispatcher->strand.post( [this, results]() {
+      boost::asio::post( my_impl->thread_pool->get_executor(), [dispatcher = my_impl->dispatcher.get(), results]() {
```

Comment:
Is it dangerous to post a lambda containing a captured pointer (dispatcher) from `my_impl` without also ensuring the lifetime of `my_impl` by capturing a `shared_ptr` to it?

Reply:
`plugin_shutdown` stops the `my_impl->thread_pool` which will destory any lambdas before the `thread_pool` or `my_impl` is destroyed.

Reply:
Great, thanks!